### PR TITLE
Implement operator<< for TokenAssociation and add unit test

### LINKED ISSUE #1332: [Good First Issue]: Implement `operator<<` for `TokenAssociation`
## 🆕🐥 First-Time Friendly

This issue is especially welcoming for people who are new to contributing to the **Hiero C++ SDK**.

We know that opening your first pull request can feel like a big step. Issues labeled **Good First Issue** are designed to make that experience easier, clearer, and more comfortable.

No prior knowledge of Hiero, Hedera, or distributed ledger technology is required.
Just a basic familiarity with C++ and Git is more than enough to get started.

---

## 👾 Description of the Issue

The `TokenAssociation` class declares a stream output operator in its header:

```
src/sdk/main/include/TokenAssociation.h
```

At line 71, the following declaration exists:

```cpp
friend std::ostream& operator<<(std::ostream& os, const TokenAssociation& record);
```

However, the corresponding implementation is **missing** from the source file:

```
src/sdk/main/src/TokenAssociation.cc
```

This means that code attempting to write a `TokenAssociation` to an output stream (e.g. `std::cout << myTokenAssociation`) will fail to link. The class already has a working `toString()` method that can be used as the body of the implementation.

The companion unit test file at:

```
src/sdk/tests/unit/TokenAssociationUnitTests.cc
```

also lacks a test for `operator<<`.

---

## 💡 Proposed Solution

Add the `operator<<` implementation in `TokenAssociation.cc`, delegating to the existing `toString()` method — exactly the same pattern used by `ExchangeRate` and `HbarTransfer`.

The change should:
- Not modify any existing logic
- Not change behavior or public APIs
- Be limited to adding the missing function body and one new test

---

## 👩‍💻 Implementation Steps (End-to-End)

- [ ] Open `src/sdk/main/src/ExchangeRate.cc` and find the `operator<<` implementation near the bottom of the file — use it as your reference
- [ ] Open `src/sdk/main/src/TokenAssociation.cc`
- [ ] Add the `operator<<` implementation at the end of the `Hiero` namespace, after the existing `toString()` function:

  ```cpp
  //-----
  std::ostream& operator<<(std::ostream& os, const TokenAssociation& record)
  {
    os << record.toString();
    return os;
  }
  ```

- [ ] Open `src/sdk/tests/unit/TokenAssociationUnitTests.cc`
- [ ] Add a test that verifies `operator<<` produces non-empty output for a populated `TokenAssociation`:

  ```cpp
  //-----
  TEST_F(TokenAssociationUnitTests, OutputStream)
  {
    // Given
    TokenAssociation tokenAssociation;
    tokenAssociation.mAccountId = getTestAccountId();
    tokenAssociation.mTokenId = getTestTokenId();

    // When
    std::ostringstream oss;
    oss << tokenAssociation;

    // Then
    EXPECT_FALSE(oss.str().empty());
  }
  ```

  Add `#include <sstream>` near the top of the test file with the other includes.

- [ ] Build the SDK and confirm all tests pass (CI will also run automatically when you open a pull request)

---

## ✅ Acceptance Criteria

- [ ] **Scope:** Changes are limited to `TokenAssociation.cc` and `TokenAssociationUnitTests.cc`
- [ ] **Behavior:** No SDK behavior or public API changes
- [ ] **Tests:** Existing tests continue to pass; new `OutputStream` test passes
- [ ] **Review:** All code review feedback addressed

---

## 📋 Step-by-Step Contribution Guide

- [ ] Comment `/assign` to request the issue
- [ ] Wait for assignment
- [ ] Fork the repository and create a branch
- [ ] Set up the project by following the instructions in `README.md`
- [ ] Make the requested changes
- [ ] Sign each commit using `-s -S`
- [ ] Push your branch and open a pull request

Read [Workflow Guide](https://github.com/hiero-ledger/hiero-sdk-cpp/blob/main/docs/training/workflow.md) for step-by-step workflow guidance.
Read [README.md](https://github.com/hiero-ledger/hiero-sdk-cpp/blob/main/README.md) for setup instructions.

❗ Pull requests **cannot be merged** without `S` and `s` signed commits. See the [Signing Guide](https://github.com/hiero-ledger/hiero-sdk-cpp/blob/main/docs/training/signing.md).

---

## 🤔 Additional Information

Reference implementation to follow:
- `src/sdk/main/src/ExchangeRate.cc` — shows the exact pattern for `operator<<` using `toString()`
- `src/sdk/main/include/TokenAssociation.h` — the declaration already exists at line 71

If you have questions while working on this issue, reach out to the @hiero-ledger/hiero-sdk-good-first-issue-support team or join the community on Discord:
[Hiero-SDK-C++](https://discord.com/channels/905194001349627914/1337424839761465364)

Maintainers are happy to help first-time contributors succeed!

### PR DESCRIPTION
**Pull request by vidhhya1**
**Description:**
Implemented the missing `operator<<` for `TokenAssociation` by delegating to the existing `toString()` method.

**Changes:**

* Added `operator<<` implementation in `TokenAssociation.cc`
* Added unit test to verify output stream behavior
* Included required headers (`<ostream>` and `<sstream>`)

**Related issue(s):**
Fixes #1332

**Notes for reviewer:**
This follows the same pattern used in similar classes and does not modify existing logic.

**Checklist:**

* [x] Documented (code comments where needed)
* [x] Tested (unit test added)

### CODE REVIEW

File: src/sdk/main/src/TokenAssociation.cc [hunk 1]
```diff
@@ -4,7 +4,8 @@
 
 #include <nlohmann/json.hpp>
 
-#include <services/custom_fees.pb.h>
+#include <services/custom_fees.pb.h>
```
Comment (rwalworth) [inline src/sdk/main/src/TokenAssociation.cc]: **issue (blocking):** A trailing space was introduced on this line. Please remove it - the clang-format check will fail otherwise.

```diff
-#include <services/custom_fees.pb.h> 
+#include <services/custom_fees.pb.h>
```

File: src/sdk/main/src/TokenAssociation.cc [hunk 2]
```diff
@@ -57,5 +58,9 @@ std::string TokenAssociation::toString() const
   json["mTokenId"] = mTokenId.toString();
   return json.dump();
 }
-
+std::ostream& operator<<(std::ostream& os, const TokenAssociation& record)
+{
+  os << record.toString();
+  return os;
+}
```
[the remark this answers is not in the document]
Comment (vidhhya1) [reply src/sdk/main/src/TokenAssociation.cc]: Thanks for the detailed review!

I have addressed the requested changes:
- Fixed the test structure (moved OutputStream test outside ToBytes test)
- Removed redundant <ostream> include
- Added separator comment for consistency
- Cleaned up formatting issues

Please let me know if anything else needs improvement.

**Review by rwalworth — APPROVED** [review]
Thanks for addressing all the feedback, @vidhhya1! Everything looks good - just left one small non-blocking suggestion in the test file. Running the workflows now and will merge once they pass!
